src/auth/middlewares/session.middleware.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable, NestMiddleware } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { InjectConnection } from "@nestjs/mongoose";
+import MongoStore from "connect-mongo";
+import { Request, Response, NextFunction, RequestHandler } from "express";
+import session from "express-session";
+import { Connection } from "mongoose";
+
+@Injectable()
+export class SessionMiddleware implements NestMiddleware {
+  private readonly requestHandler: RequestHandler;
+  constructor(
+    private readonly configService: ConfigService,
+    @InjectConnection() private readonly mongoConnection: Connection,
+  ) {
+    let store = {};
+    if (this.configService.get<string>("expressSession.store") === "mongo")
+      store = MongoStore.create({
+        client: this.mongoConnection.getClient(),
+        ttl: this.configService.get<number>("jwt.expiresIn"),
+      });
+    this.requestHandler = session({
+      secret: this.configService.get<string>("expressSession.secret") as string,
+      resave: false,
+      saveUninitialized: true,
+      ...store,
+    });
```

Comment:
I tested it locally, and the session method does not register the mongoStore.
It appears to me that session function cannot handle `...store`
The correct type should be `store?: Store | undefined;`, you can check the type from the package interface.

```suggestion
    let store = Store | undefined;
    if (this.configService.get<string>("expressSession.store") === "mongo")
      store = MongoStore.create({
        client: this.mongoConnection.getClient(),
        ttl: this.configService.get<number>("jwt.expiresIn"),
      });
    this.requestHandler = session({
      secret: this.configService.get<string>("expressSession.secret") as string,
      resave: false,
      saveUninitialized: true,
      store,
    });
```

Reply:
thanks for noticing this, I made a mistake indeed in the latest commit. I'll fix in a min PR and tests!

Reply:
The test looks fine to me. Maybe one additional API test to mock the process and see if user session is created in the db?

Reply:
the test was also wrong, see the new commit. These controller test is a good idea I think, I did not add it before because all API tests are done with the /test folder from where I cannot check the mongo session. But I can add one indeed in the auth.controller.spec.ts

Reply:
You can run `db.collection("sessions").find({})` under the /test for checking I think, it's been done in many API tests. I think having a spec.ts is also fine, but maybe take more efforts

Reply:
ah nice, thanks for pointing out I did not see that! Will do

Reply:
done with jest as it required mocking the OidcGuard